### PR TITLE
Add change listener to main table to scroll to imported entry

### PR DESCRIPTION
**Pull request by b0n541**
Add change listener to main table to scroll to imported entry #5383

----

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes
      --> not really testable via unit tests
- [x] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
     --> not needed
- [x] Ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)

### CODE REVIEW

File: src/main/java/org/jabref/gui/maintable/MainTable.java [hunk 1]
```diff
@@ -112,6 +113,15 @@ public MainTable(MainTableDataModel model, JabRefFrame frame,
         // Enable sorting
         model.getEntriesFilteredAndSorted().comparatorProperty().bind(this.comparatorProperty());
 
+        model.getEntriesFilteredAndSorted().addListener(
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/maintable/MainTable.java]: Although it is a neat idea to use the listener, I'm afraid that it does not work as intended. The listener is also triggered if I clear a search as then a bunch of previously filtered-out entries are readded to the list. What should work is if you add a listener to `database.getDatabase()`.

File: src/main/java/org/jabref/gui/maintable/MainTable.java [hunk 2]
```diff
@@ -112,6 +113,15 @@ public MainTable(MainTableDataModel model, JabRefFrame frame,
...
+            (ListChangeListener<BibEntryTableViewModel>) change -> {
+                if (change.next()) {
+                    if (change.wasAdded()) {
```
Comment (tobiasdiez) [inline src/main/java/org/jabref/gui/maintable/MainTable.java]: Could you please also add a `!change.getAddedSubList().isEmpty()` check to make sure that the code below executes without exception (that test might be covered by `wasAdded` but I'm not sure)

File: CHANGELOG.md [hunk 3]
```diff
@@ -38,7 +38,7 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - We fixed a problem where the "editor" information has been duplicated during saving a .bib-Database. [#5359](https://github.com/JabRef/jabref/issues/5359)
 - We re-introduced the feature to switch between different preview styles. [#5221](https://github.com/JabRef/jabref/issues/5221)
 - We fixed various issues (including [#5263](https://github.com/JabRef/jabref/issues/5263)) related to copying entries to the clipboard
-
+- We added a change listener to the main table to scroll to imported entry [#5383](https://github.com/JabRef/jabref/issues/5383)
```
Comment (tobiasdiez) [inline CHANGELOG.md]: The change log is mostly for our users and most of them don't know what a "change listener" is ;-). Please reformulate this a bit more user-friendly.

File: src/main/java/org/jabref/gui/maintable/MainTable.java (hunk 2 above)
Comment (b0n541) [reply src/main/java/org/jabref/gui/maintable/MainTable.java]: It's covered by `wasAdded()`.

File: src/main/java/org/jabref/gui/maintable/MainTable.java (hunk 1 above)
Comment (b0n541) [reply src/main/java/org/jabref/gui/maintable/MainTable.java]: I ran into multiple race conditions after I added a change listener to the internal observable list of bib entries in the database. Therefore I choose to introduce a new event that is sent over the event bus when the insertion of one or many entries is finished.

File: CHANGELOG.md (hunk 3 above)
Comment (b0n541) [reply CHANGELOG.md]: Done.

**Comment by b0n541** [comment]
Please re-review.

**Comment by tobiasdiez** [comment]
@Siedlerchr what do you think?

**Comment by Siedlerchr** [comment]
Loos good to me now as well! Thanks for your contribution

**Pull request merged**